measureme/src/profiler.rs

```diff
@@ -111,4 +113,46 @@ impl<S: SerializationSink> Profiler<S> {
                 bytes.copy_from_slice(raw_event_bytes);
             });
     }
+
+    /// Creates a "start" event and returns a `TimingGuard` that will create
+    /// the corresponding "end" event when it is dropped.
+    pub fn start_event<'a>(
+        &'a self,
+        event_kind: StringId,
+        event_id: StringId,
+        thread_id: u64,
+    ) -> TimingGuard<'a, S> {
+        self.record_event(event_kind, event_id, thread_id, TimestampKind::Start);
+
+        TimingGuard(TimingGuardInternal {
+            profiler: self,
+            event_id,
+            event_kind,
+            thread_id,
+        })
+    }
+}
+
+/// When dropped, this `TimingGuard` will record an "end" event in the
+/// `Profiler` it was created by.
+#[must_use]
+pub struct TimingGuard<'a, S: SerializationSink>(TimingGuardInternal<'a, S>);
+
+struct TimingGuardInternal<'a, S: SerializationSink> {
```

Comment:
I just realized that the `TimingGuard`/`TimingGuardInternal` split is not necessary for measureme. Let me fix that...